deepchecks/tabular/checks/data_integrity/feature_feature_correlation.py

```diff
@@ -0,0 +1,96 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""module contains  Feature-Feature Correlation check."""
+
+import pandas as pd
+import plotly.express as px
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.utils.correlation_methods import correlation_ratio, symmetric_theil_u_correlation
+from deepchecks.utils.dataframes import generalized_corrwith
+
+__all__ = ['FeatureFeatureCorrelation']
+
+
+class FeatureFeatureCorrelation(SingleDatasetCheck):
+    """
+    Checks for pairwise correlation between the features.
+
+    Extremely correlated pairs could indicate redundancy and even duplication.
+    """
+
+    def __init__(self,
+                 **kwargs):
+        super().__init__()
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """
+        Run Check.
+
+        Returns
+        -------
+        CheckResult
+            A DataFrame of the pairwise correlations between the features.
+        """
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+
+        dataset.assert_features()
+        num_features = dataset.numerical_features
+        cat_features = dataset.cat_features
+
+        encoded_cat_data = dataset.data.loc[:, cat_features].apply(lambda x: pd.factorize(x)[0])
+
+        all_features = num_features + cat_features
+        full_df = pd.DataFrame(index=all_features, columns=all_features)
+
+        # Numerical-numerical correlations
+        if num_features:
+            full_df.loc[num_features, num_features] = dataset.data.loc[:, num_features].corr(method='spearman')
+
+        # Categorical-categorical correlations
+        if cat_features:
+            full_df.loc[cat_features, cat_features] = encoded_cat_data.corr(method=symmetric_theil_u_correlation)
+
+        # Numerical-categorical correlations
+        if num_features and cat_features:
+            num_cat_corr = generalized_corrwith(dataset.data.loc[:, num_features], encoded_cat_data,
```

Comment:
IMO places in which the numeric feature has a null value should be ignored (masked)

Reply:
the way null values are handled depends on the method